pkg/schema/v1/contracts.go

```diff
@@ -10,24 +11,32 @@ import (
 	"reflect"
 )
 
+// Private type to prevent collisions with other context keys
+type contextKey string
```

Comment:
I would move the whole context related stuff to `pkg/cluster` so that it is encapsulated.